src/Scripting/ScriptingSystem.cpp

```diff
@@ -54,16 +56,40 @@ void ScriptingSystem::_initBaseLibraries() {
     );
 }
 
+/* Internal lua function used as package loader for the Bindings table.
+* 
+* Usage in lua: 
+*   local gameBindings = require "bindings.game" -- If the module starts with 'bindings.' we try to load/create the binding table
+*   gameBindings.doSomething()
+* 
+* TODO(Gerark) I'm asking in the sol2 repo if there's a way to avoid a lua_CFunction and use the sol2 approach instead.
+* Here's the question: https://github.com/ThePhD/sol2/issues/1601 */
+int _loadBindingTableThroughRequire(lua_State *luaState) {
+    std::string path = sol::stack::get<std::string>(luaState, 1);
+    std::string_view prefix = "bindings.";
+    if (path.starts_with(prefix)) {
+        // When requesting a module, Lua expects us to place a code chunk on the stack.
+        // We utilize loadbuffer to load this chunk, after which the Lua VM promptly executes it.
+        std::string script = std::format("return _createBindingTable('{}')", path);
+        luaL_loadbuffer(luaState, script.data(), script.size(), path.c_str());
+        return 1;
+    }
+    return 0;
+}
```

Comment:
This code ended up being a bit more uglier than before but the improvement on the lua side justify it imo.